### PR TITLE
updated .gitignore with gitignore.io snippet

### PR DESCRIPTION
**Pull request by kwk**
With https://www.gitignore.io/?templates=go,vim,git,macos,linux,emacs,eclipse,windows,visualstudiocode we have a nice and easy way to manage out git ignore file.

The pieces that are relevant just for this project appear at the top of the `.gitignore` file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,146 @@
-### IntelliJ ###
-.idea/
```
Comment (corinnekrych) [inline .gitignore]: @kwk would you mind leaving the `.idea/` for the goland IDE in the `.gitignore`
I see you adding other IDE which is fine but let's leave that one

Comment (kwk) [reply .gitignore]: Why do we want to put any IDE setting under version control? I don't see why we should do that?

Comment (corinnekrych) [reply .gitignore]: No
Like you ignore eclipse, emac, ignore also goland idea

Comment (kwk) [reply .gitignore]: Ah :) sorry, I misread the comment as: "please add it". On it.

Comment (kwk) [reply .gitignore]: I've updated the gitignore.io URL to include intellij: https://www.gitignore.io/?templates=go,vim,git,macos,linux,emacs,windows,eclipse,intellij+all,visualstudiocode Are you fine with this @corinnekrych ?

**Pull request merged**